pkg/receive/handler.go

```diff
@@ -325,7 +325,7 @@ func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 		tenant = h.options.DefaultTenantID
 	}
 
-	// TODO(yeya24): handle remote write metadata and exemplars.
+	// TODO(yeya24): handle remote write metadata.
 	// exit early if the request contained no data
 	if len(wreq.Timeseries) == 0 {
```

Comment:
This should be updated as we need to process exemplars only requests.

Reply:
This is working as is, as exemplars are part of the `Timeseries` array. This is why we don't need any modification on the forwarding part as well.

Reply:
You can take a look at the proto definition here: https://github.com/thanos-io/thanos/blob/main/pkg/store/storepb/prompb/types.proto#L66-L72

Since we batch and forward the Timeseries elements, the exemplars get forwarded automatically.

Reply:
Got it. If it is part of the `Timeseries` array then it is good here.

I am not sure about the forwarding part. How does the hashing part work https://github.com/thanos-io/thanos/blob/main/pkg/receive/handler.go#L388?

https://github.com/thanos-io/thanos/blob/main/pkg/receive/hashring.go#L65 Hashring.GetN calculates the hash based on `ts.Labels`. If it is an exemplar series, what will be the value of `ts.Labels`?  Is it the series labels of exemplar?

Reply:
Yes, that makes sense. If it is part of `timeseries` then all good. Thanks.